apps/dashboard/src/components/workflow-editor/workflow-canvas.tsx

```diff
@@ -57,6 +56,33 @@ const panOnDrag = [1, 2];
 // y distance = node height + space between nodes
 const Y_DISTANCE = NODE_HEIGHT + 50;
 
+const mapStepToNodeContent = (step: Step): string | undefined => {
+  const controlValues = step.controls.values;
+
+  switch (step.type) {
+    case StepTypeEnum.TRIGGER:
+      return 'This step triggers this workflow';
+    case StepTypeEnum.EMAIL:
+      return 'Sends Email to your subscribers';
+    case StepTypeEnum.SMS:
+      return 'Sends SMS to your subscribers';
+    case StepTypeEnum.IN_APP:
+      return 'Sends In-App notification to your subscribers';
+    case StepTypeEnum.PUSH:
+      return 'Sends Push notification to your subscribers';
+    case StepTypeEnum.CHAT:
+      return 'Sends Chat message to your subscribers';
+    case StepTypeEnum.DELAY:
+      return `Delay for ${controlValues.amount} ${controlValues.unit}`;
+    case StepTypeEnum.DIGEST:
+      return 'Batches events into one coherent message before delivery to the subscriber.';
+    case StepTypeEnum.CUSTOM:
+      return 'Executes the business logic in your bridge application';
+    default:
+      return undefined;
+  }
+};
```

Comment:
Added a single place where node content is being created that has access to all step data.

We can also think about adding control values data to other steps, e.g. digest.